geoh5py/groups/property_group_table.py

```diff
@@ -0,0 +1,162 @@
+            for idx, key in enumerate(self.locations_keys):
+                output_array[key] = self.locations[:, idx]
+
+        for key, name in zip(keys, names, strict=False):
+            data = self.property_group.parent.get_data(key)[0]
+            output_array[str(name)] = data.values
+
+        return output_array
+
+    def _create_empty_structured_array(
+        self,
+        properties_name: list[str] | list[UUID],
+        properties_keys: list[UUID],
+        spatial_index: bool = False,
+    ) -> np.ndarray:
+        """
+        Create an empty structured array that can contains the data.
+
+        :param properties_name: The names of the properties.
+        :param properties_keys: The keys of the properties.
+        :param spatial_index: If True, the association is added to the table.
+
+        :return: an empty structured array.
+        """
+        dtypes = (
+            [(loc, np.float32) for loc in self.locations_keys] if spatial_index else []
+        )
+
+        for key, name in zip(properties_keys, properties_name, strict=False):
+            data = self.property_group.parent.get_data(key)[0]
+            dtype: type | str = DataTypeEnum.from_primitive_type(
+                data.entity_type.primitive_type
+            )
+            if dtype not in [np.float32, np.int32, np.uint32, bool]:
+                dtype = "O"
+            dtypes.append((str(name), dtype))
+
+        empty_array = np.recarray((self.size,), dtype=dtypes)
+
+        return empty_array
+
+    @property
+    def locations(self) -> np.ndarray:
+        """
+        The locations of the association table.
+
+        This function is needed as a data can be both associated to cell or
+        vertex in CellObjects.
+        """
+        if self.property_group.association == DataAssociationEnum.VERTEX:
+            return self.property_group.parent.vertices
+
+        if self.property_group.association == DataAssociationEnum.CELL:
+            return self.property_group.parent.centroids  # type: ignore
+
+        raise ValueError(
+            f"The association {self.property_group.association} is not supported. "
+            f"Only VERTEX and CELL associations are supported."
+        )
+
+    @property
+    def property_group(self) -> PropertyGroup:
+        """
+        The property group to extract the data from.
+        """
+        return self._property_group
+
+    @property
+    def size(self) -> int:
+        """
+        The size of the properties in the group.
+        """
+        if self.property_group.association == DataAssociationEnum.VERTEX:
+            return self.property_group.parent.n_vertices
+
+        if self.property_group.association == DataAssociationEnum.CELL:
+            return self.property_group.parent.n_cells
+
+        raise ValueError(
+            f"The association {self.property_group.association} is not supported. "
+            f"Only VERTEX and CELL associations are supported."
+        )
```

Comment:
```suggestion
        return self.locations.shape[0]
```

Reply:
Cell object can have data with different associations:
Cell and Vertex. 
We have to deal with this scenario.